content/sensu-go/5.17/reference/secrets.md

```diff
@@ -0,0 +1,214 @@
+
+namespace    |      |
+-------------|------
+description  | [Sensu RBAC namespace][9] that the secret belongs to.
+required     | true
+type         | String
+example      | {{< highlight shell >}}"namespace": "default"{{< /highlight >}}
+
+### Spec attributes
+
+id           | 
+-------------|------ 
+description  | The identifying key for the provider to retrieve the secret.
+required     | true
+type         | String
+example      | {{< highlight shell >}}"id": "secret/ansible#token"{{< /highlight >}}
+
+provider     | 
+-------------|------ 
+description  | The name of the Sensu provider with the secret.
+required     | true
+type         | String
+example      | {{< highlight shell >}}"provider": "ansible_vault"{{< /highlight >}}
+
+## Secret configuration
+
+Use the [Secrets API][2] and [sensuctl][3] to create, view, and manage your secrets configuration.
+To manage secrets configuration with sensuctl, configure sensuctl as the default [`admin` user][6].
+
+The [standard sensuctl subcommands][4] are available for secrets (list, info, and delete).
+
+To list all secrets:
+
+{{< highlight shell >}}
+sensuctl secret list
+{{< /highlight >}}
+
+To see a secret's status:
+
+{{< highlight shell >}}
+sensuctl secret info SECRET_NAME
+{{< /highlight >}}
+
+To delete a secret:
+
+{{< highlight shell >}}
+sensuctl secret delete SECRET_NAME
+{{< /highlight >}}
+
+## Secret examples
+
+A secret resource definition refers to a secrets `id` and a secrets `provider`.
+Read the [secrets provider reference][7] for the provider specification.
+
+{{< language-toggle >}}
+
+{{< highlight yml >}}
+---
+type: Secret
+api_version: secrets/v1
+metadata:
+  name: sensu-ansible-token
+  namespace: default
+spec:
+  id: ANSIBLE_TOKEN
+  provider: env
+{{< /highlight >}}
+
+{{< highlight json >}}
+{
+  "type": "Secret",
+  "api_version": "secrets/v1",
+  "metadata": {
+    "name": "sensu-ansible-token",
+    "namespace": "default"
+  },
+  "spec": {
+    "id": "ANSIBLE_TOKEN",
+    "provider": "env"
+  }
+}
+{{< /highlight >}}
+
+{{< /language-toggle >}}
+
+Secrets that target a HashiCorp Vault always have an `id` as shown in the following example:
+
+{{< language-toggle >}}
+
+{{< highlight yml >}}
+---
+type: Secret
+api_version: secrets/v1
+metadata:
+  name: sensu-ansible-token
+  namespace: default
+spec:
+  id: 'secret/website#database'
+  provider: ansible_vault
+{{< /highlight >}}
+
+{{< highlight json >}}
+{
+  "type": "Secret",
+  "api_version": "secrets/v1",
+  "metadata": {
+    "name": "database",
+    "namespace": "default"
+  },
+  "spec": {
+    "id": "secret/website#database",
+    "provider": "vault"
+  }
+}
+{{< /highlight >}}
+
+{{< /language-toggle >}}
+
+Secrets that target a HashiCorp Vault must start with the word `secret`.
+In this example, the name of the secret is `website`.
+The `website` secret contains a value called `database`, which is the password to our database.
```

Comment:
This example might make more sense if the secret was `secret/database#password`, because due to the Vault API, you can store multiple fields in a keyspace such as `secret/database#url` and `secret/database#username` for example.